### PR TITLE
Iter23

### PR DESCRIPTION
**Pull request by kripsy**

### CODE REVIEW

**Comment by kripsy** [comment]
@arttet 
не успел разобраться почему не отображается репорт от codecover.
завтра разберусь, сейчас просто отправлю ссылку, чтобы практикум посчитал закрытие спринта.

**Comment by kripsy** [comment]
@arttet поправил codecover))

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,5 @@ vendor/
 # IDEs directories
 .idea
 .vscode
+pki/*.crt
```
Comment (arttet) [inline .gitignore]: Maybe cert?
```suggestion
cert/*.crt
```

File: cmd/shortener/main.go [hunk 2]
```diff
@@ -71,11 +84,128 @@ func main() {
...
+	go func() {
+		<-sigint
+		if err := srv.Shutdown(context.Background()); err != nil {
+			fmt.Fprintf(os.Stderr, "error shotdown server: %v\n", err)
+		}
+		close(connsClosed)
+	}()
+
+	if application.GetAppConfig().EnableHTTPS != "" {
+		fmt.Println("CREATE CERT")
```
Comment (arttet) [inline cmd/shortener/main.go]: It would be better if it is debug log

File: cmd/shortener/main.go [hunk 3]
```diff
@@ -71,11 +84,128 @@ func main() {
...
+		err = createCertificate()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error: %v\n", err)
+
+			return
+		}
+		fmt.Println("Success CREATE CERT")
+		err = srv.ListenAndServeTLS("./pki/server.crt", "./pki/server.key")
```
Comment (arttet) [inline cmd/shortener/main.go]: This is a magic string

File: cmd/shortener/main.go [hunk 4]
```diff
@@ -71,11 +84,128 @@ func main() {
...
+		//nolint:gomnd
+		IdleTimeout: 30 * time.Second,
+		//nolint:gomnd
+		ReadHeaderTimeout: 2 * time.Second,
+		Handler:           application.GetAppServer().Router,
+	}
+
+	go func() {
+		<-sigint
+		if err := srv.Shutdown(context.Background()); err != nil {
```
Comment (arttet) [inline cmd/shortener/main.go]: Please use context with timeout

File: cmd/shortener/main.go [hunk 5]
```diff
@@ -71,11 +84,128 @@ func main() {
...
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 
 		return
 	}
+
+	<-connsClosed
+	fmt.Println("Server Shutdown successfully")
+}
+
+func createCertificate() error {
```
Comment (arttet) [inline cmd/shortener/main.go]: It should not be in `main`

File: internal/app/logger/logger_test.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
+// Package logger provides functionality for logger.
+//
+//nolint:testpackage
```
Comment (arttet) [inline internal/app/logger/logger_test.go]: Why?

File: internal/app/config/config.go [hunk 7]
```diff
@@ -91,17 +137,82 @@ func InitConfig() *Config {
...
+	cfg := map[string]interface{}{}
+	err = json.Unmarshal(data, &cfg)
+	if err != nil {
+		fmt.Printf("error unmarshall config %v: ", err)
+
+		return "", "", "", "", "", fmt.Errorf("error unmarshall config %w", err)
+	}
+
+	for k, v := range inputParams {
+		//nolint:nestif
```
Comment (arttet) [inline internal/app/config/config.go:190]: This is a good point

File: cmd/shortener/main.go [hunk 8]
```diff
@@ -71,11 +84,128 @@ func main() {
...
+	connsClosed := make(chan struct{})
+	sigint := make(chan os.Signal, 1)
+	signal.Notify(sigint, os.Interrupt)
+
+	srv := &http.Server{
+		Addr:         application.GetAppConfig().URLServer,
+		ReadTimeout:  time.Second,
+		WriteTimeout: time.Second,
+		//nolint:gomnd
+		IdleTimeout: 30 * time.Second,
```
Comment (arttet) [inline cmd/shortener/main.go]: This is a magic number

**Review by arttet — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (kripsy) [reply .gitignore]: fixed

File: cmd/shortener/main.go (hunk 2 above)
Comment (kripsy) [reply cmd/shortener/main.go]: fixed

File: cmd/shortener/main.go (hunk 3 above)
Comment (kripsy) [reply cmd/shortener/main.go]: fixed

File: cmd/shortener/main.go (hunk 4 above)
Comment (kripsy) [reply cmd/shortener/main.go]: fixed

File: internal/app/logger/logger_test.go (hunk 6 above)
Comment (kripsy) [reply internal/app/logger/logger_test.go]: fixed for all tests

File: cmd/shortener/main.go (hunk 8 above)
Comment (kripsy) [reply cmd/shortener/main.go]: fixed

File: internal/app/config/config.go (hunk 7 above)
Comment (kripsy) [reply internal/app/config/config.go:190]: I wasn't sure about this. imho))

File: cmd/shortener/main.go (hunk 5 above)
Comment (kripsy) [reply cmd/shortener/main.go]: fixed

**Comment by kripsy** [comment]
@arttet doooooneeee 🫡

**Comment by kripsy** [comment]
@arttet smth anything else?))

**Comment by arttet** [comment]
> @arttet smth anything else?))

Please create a ticket

**Comment by kripsy** [comment]
> > @arttet smth anything else?))
> 
> Please create a ticket

Ticket about what?
Linters? I don't think that warning is a some mistake in product. It's all my responsibility, because I edit config and enable all linters.

**Comment by arttet** [comment]
> > > @arttet smth anything else?))
> > 
> > Please create a ticket
> 
> Ticket about what?
> Linters? I don't think that warning is a some mistake in product. It's all my responsibility, because I edit config and enable all linters.

Please create a ticket in the tracking system.

**Comment by kripsy** [comment]
> > > > @arttet smth anything else?))
> > > 
> > > 
> > > Please create a ticket
> > 
> > 
> > Ticket about what?
> > Linters? I don't think that warning is a some mistake in product. It's all my responsibility, because I edit config and enable all linters.
> 
> Please create a ticket in the tracking system.

I hope you mean a new ticket for checking my task in the Yandex prakticum.)) 🤔

File: Makefile [hunk 9]
```diff
@@ -37,5 +37,9 @@ migration_force:
 run_autotest: build
 	~/go/shortenertest-darwin-arm64 -test.v -test.run=TestIteration1 -binary-path=./cmd/shortener/shortener -source-path=./
 
+.PHONY: run_unittest
```
Comment (arttet) [inline Makefile:40]: Maybe: `test` ?

Comment (arttet) [inline Makefile:40]:
```suggestion
.PHONY: test
```

File: codecov.yml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+coverage:
```
Comment (arttet) [inline codecov.yml:1]:
```suggestion
coverage:
  range: 80..100
  round: down
  precision: 2
```

**Review by arttet — APPROVED** [review]

**Comment by kripsy** [comment]
@arttet u didn't accept in tracking my task)) pls accept it in Yandex practikum))

**Pull request merged**